lib/routemap.c

```diff
@@ -1649,7 +1660,9 @@ void route_map_event_hook(void (*func)(route_map_event_t, const char *))
 /* Routines for route map dependency lists and dependency processing */
 static bool route_map_rmap_hash_cmp(const void *p1, const void *p2)
 {
-	return (strcmp((const char *)p1, (const char *)p2) == 0);
+	return (strcmp(((const struct route_map_dep_data *)p1)->rname,
```

Comment:
Please fix this function appropriately.  This is a crash waiting to happen.

Reply:
Thanks, @donaldsharp.
I've addressed this.